lib/errors/error-collector.js

```diff
@@ -142,13 +142,18 @@ class ErrorCollector {
     }
 
     const isErroredTransaction = urltils.isError(this.config, transaction.statusCode)
+    const isIgnoredErrorStatusCode = urltils.isIgnoredError(
+      this.config,
+      transaction.statusCode
+    )
+
     const isExpectedErrorStatusCode = urltils.isExpectedError(
       this.config,
       transaction.statusCode
     )
 
     // collect other exceptions only if status code is not ignored
-    if (transaction.exceptions.length) {
+    if (transaction.exceptions.length && !isIgnoredErrorStatusCode) {
```

Comment:
This was the main part of the bug. Basically restored some code prior to the later refactors.

Reply:
Alright, I think I've grokked it. It looks like the `shouldIgnoreError()` function (called when an error is added in `add()`) looks at the transaction's `statusCode` to test whether it should be ignored. But by the time the transaction has ended the `statusCode` may have changed. When `onTransactionEnd()` is called, the `statusCode` will be set to its final value. It calls `collect()`, which use to do ignore checks, but due to refactors was moved out. So, this will stop the collection of all add exceptions if the status code should be ignored.